RDKShell/RDKShell.cpp

```diff
@@ -7162,6 +7243,281 @@ namespace WPEFramework {
 
             return result;
         }
+
+        bool RDKShell::MemCheckpointRestoreClient::checkpoint(const std::string &callSign, uint32_t timeouteMs)
+        {
+            bool isCheckpointAllowed = false;
+
+            mProcessedAppsLock.lock();
+            if(mProcessedApps.find(callSign) == mProcessedApps.end())
+            {
+                //App not processed
+                isCheckpointAllowed = true;
+            }
+            else if(mProcessedApps[callSign] == RDKShell::MemCheckpointRestoreClient::RESTORING)
+            {
+                //App being processeed, then shedule checkpoint only if restore is ongoing
+                isCheckpointAllowed = true;
```

Comment:
I don't understand how we can proceed with checkpoint() operation when the app is in "RESTORING" state ?

Reply:
Memcr is handling requests one by one. From app cycle perspective it may make some sense to trigger Checkpoint if Restore is still ongoing, since after Restore the App should be still suspended

Reply:
Now I am thinking if above won't be overcomplicated for ie RA logic. And for simplicity, it is better to remove this Checkpoint condition